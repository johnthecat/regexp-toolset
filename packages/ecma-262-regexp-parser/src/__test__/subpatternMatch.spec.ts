import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../index.js';

describe('Subpattern Match', () => {
  it('should refer to group with same name', () => {
    expect(parseRegexp(/(?<my_name>a) \k<my_name>/)).toMatchSnapshot();
  });

  it('should throw if there is no group with such name', () => {
    expect(() => parseRegexp('/\\k<my_name>/')).toThrowErrorMatchingInlineSnapshot(`
      "
       ❱ /\\\\k<my_name>/
          ═══════════
       This token references a non-existent or invalid subpattern"
    `);
  });
});
