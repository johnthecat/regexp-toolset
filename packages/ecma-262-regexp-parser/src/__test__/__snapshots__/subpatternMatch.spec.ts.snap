// Vitest Snapshot v1

exports[`Subpattern Match > should refer to group with same name 1`] = `
{
  "body": {
    "end": 25,
    "expressions": [
      {
        "body": {
          "charCode": 97,
          "end": 12,
          "kind": 6,
          "start": 12,
          "type": "simple",
          "value": "a",
        },
        "end": 13,
        "kind": 23,
        "specifier": {
          "end": 11,
          "kind": 24,
          "name": "my_name",
          "start": 3,
        },
        "start": 1,
        "type": "capturing",
      },
      {
        "charCode": 32,
        "end": 14,
        "kind": 6,
        "start": 14,
        "type": "simple",
        "value": " ",
      },
      {
        "end": 25,
        "groupName": "my_name",
        "kind": 11,
        "ref": {
          "body": {
            "charCode": 97,
            "end": 12,
            "kind": 6,
            "start": 12,
            "type": "simple",
            "value": "a",
          },
          "end": 13,
          "kind": 23,
          "specifier": {
            "end": 11,
            "kind": 24,
            "name": "my_name",
            "start": 3,
          },
          "start": 1,
          "type": "capturing",
        },
        "start": 15,
      },
    ],
    "kind": 5,
    "start": 1,
  },
  "end": 26,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;
